recipes/apr-util/all/conanfile.py

```diff
@@ -95,16 +98,20 @@ def requirements(self):
         if self.options.with_mysql:
             self.requires("libmysqlclient/8.0.17")
         if self.options.with_sqlite3:
-            self.requires("sqlite3/3.31.1")
+            self.requires("sqlite3/3.35.1")
         if self.options.with_expat:
-            self.requires("expat/2.2.9")
+            self.requires("expat/2.2.10")
         if self.options.with_postgresql:
-            self.requires("libpq/11.5")
+            self.requires("libpq/13.2")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
 
+    def validate(self):
+        if self.options.shared != self.options["apr"].shared:
+            raise ConanInvalidConfiguration("apr-util must be built with same shared option as apr")
```

Comment:
@madebr, this is why forcing upstream option in configure() is useful, here shared apr-util is not tested.

Reply:
Yes, but that can override user's options if it is built with `-o apr:shared=False -o apr-util=True`.
You can override the option if you absolutely know that the option is not passed, but I think this behavior can get complex and hard to understand fast.

Reply:
Yes, it's an inconsistency in conan side. Overriding rules are not the same depending if upstream options are set through command lines or a conanfile.py.